src/som/interpreter/SArguments.java

```diff
@@ -58,4 +91,139 @@ public static SImmutableArray getArgumentsWithoutReceiver(final Object[] argumen
     }
     return result;
   }
+
+  public static Object[] getPlainArgumentsWithReceiver(final Object receiver,
+      final SArray args, final SizeAndLengthPrim size, final AtPrim at,
+      final ExpressionNode expression,
+      final ShadowStackEntryLoad entryLoad,
+      final VirtualFrame frame) {
+    int argSize = (int) (size.executeEvaluated(args) + 1);
+    int defaultArgSize = argSize;
+    if (VmSettings.ACTOR_ASYNC_STACK_TRACE_STRUCTURE) {
+      argSize++;
+    }
+
+    Object[] result = new Object[argSize];
+    result[0] = receiver;
+    for (int i = 1; i < defaultArgSize; i++) {
+      result[i] = at.executeEvaluated(null, args, (long) i);
+    }
+
+    if (VmSettings.ACTOR_ASYNC_STACK_TRACE_STRUCTURE) {
+      entryLoad.loadShadowStackEntry(result, expression, frame, false);
+    }
+    return result;
+  }
+
+  public static Object[] getPlainNoArgumentsWithReceiver(final Object receiver,
+      final ExpressionNode expression,
+      final ShadowStackEntryLoad entryLoad,
+      final VirtualFrame frame) {
+    if (VmSettings.ACTOR_ASYNC_STACK_TRACE_STRUCTURE) {
+      Object[] arguments = new Object[] {receiver, null};
+      entryLoad.loadShadowStackEntry(arguments, expression, frame, false);
+      return arguments;
+    } else {
+      return new Object[] {receiver};
+    }
+  }
+
+  public static Object[] getPlain1ArgumentWithReceiver(final Object receiver,
```

Comment:
In Java, we got method overloading for better or for worse. So, I don't think we need to encode arity in method names here.

Reply:
So I changed to getPlainXArgumentWithReceiver with variable number of remaining parameters and made a single function instead of 3.